### PR TITLE
Make Trin compilable/(not crash) on Windows + Add check CI

### PR DESCRIPTION
**Pull request by KolbyML**
### What was wrong?
Certain parts of Trin either won't compile on Windows or crash because rust doesn't support windows Unix IPC
### How was it fixed?
By adding ``#![cfg(unix)]`` in files that should only be compile on windows and by adding an error if someone tries to use IPC on Windows.

Reth added ``#[cfg(unix)]`` a few months back so the reth-ipc library will compile it will just crash when we try to use ipc functions.

I also added a CI using ``check`` instead of ``build``
https://doc.rust-lang.org/cargo/commands/cargo-check.html

TL:DR it has all the CI benefits of building but without the codegen part. So it is a lot quicker for the result we want.

### CODE REVIEW

**Comment by KolbyML** [comment]
I think in the future I will change the CI test from a pure windows one to one on linux doing a check for windows. I am not 100 sure if I can do it yet, but I think it would result in lower CI times so I am interested

File: portalnet/src/storage.rs [hunk 1]
```diff
@@ -1140,6 +1140,8 @@ pub mod test {
         // The restarted store should have the same radius as the original
         assert_eq!(radius, new_storage.radius);
 
+        drop(storage);
+        drop(new_storage);
```
Comment (carver) [inline portalnet/src/storage.rs:1144]: Why were the explicit drops necessary?

File: src/main.rs [hunk 2]
```diff
@@ -10,6 +12,14 @@ use trin::run_trin;
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     init_tracing_logger();
     let trin_config = TrinConfig::from_cli();
+
+    // Placing this check here since TrinConfig has tests which this would fail.
+    // We also don't want to run unneeded code if we are going to panic for this anyways.
+    #[cfg(windows)]
```
Comment (carver) [inline src/main.rs:18]: I had to read the comment a few times to understand it. Maybe something like?
```suggestion
    // Panic early on a windows build that is trying to use IPC, which is unsupported for now
    // Make sure not to panic on non-windows configurations.
    #[cfg(windows)]
```

**Review by carver — COMMENTED** [review]
What do you think about moving the check to panic on Windows to the beginning of `run_trin`?

**Comment by carver** [comment]
> I also added a CI using `check` instead of `build`
> https://doc.rust-lang.org/cargo/commands/cargo-check.html

Yeah, this makes a lot of sense. I'm even wondering if we should switch the `build` job to use check too. What do you think? 👍🏻 if you want to add it

File: portalnet/src/storage.rs (hunk 1 above)
Comment (KolbyML) [reply portalnet/src/storage.rs:1144]: They are in the tests below and above. This change isn't related to the PR. Without these drops the test will panic on certain operating systems. If you read the doc for .close() on the line before there is a comment in the example about this.

**Comment by KolbyML** [comment]
> What do you think about moving the check to panic on Windows to the beginning of run_trin?

resolved

> Yeah, this makes a lot of sense. I'm even wondering if we should switch the build job to use check too. What do you think? 👍🏻 if you want to add it

I am fine with that, would that belong in a different PR then?

File: src/main.rs (hunk 2 above)
Comment (KolbyML) [reply src/main.rs]: resolved

**Comment by ogenev** [comment]
If I understand this correctly, the claim here is that windows does not support IPC? If this is the case, I'm curious what is this support for: https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/ ? Also saw this after googling: https://batsov.com/articles/2022/01/20/unix-sockets-are-now-supported-on-windows/

**Comment by KolbyML** [comment]
> If I understand this correctly, the claim here is that windows does not support IPC? If this is the case, I'm curious what is this support for: https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/ ? Also saw this after googling: https://batsov.com/articles/2022/01/20/unix-sockets-are-now-supported-on-windows/

Rust doesn't support Windows UDS.

I am working on a PR to change this

**Comment by KolbyML** [comment]
> If I understand this correctly, the claim here is that windows does not support IPC? If this is the case, I'm curious what is this support for: https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/ ? Also saw this after googling: https://batsov.com/articles/2022/01/20/unix-sockets-are-now-supported-on-windows/

Rust doesn't support Windows UDS.

I am working on a PR to change this https://github.com/tokio-rs/mio/pull/1667

You asked the exact same question in my last support Windows PR so instead of answering it here I will link to the time I already answered this https://github.com/ethereum/trin/pull/739#discussion_r1196020473

**Review by ogenev — APPROVED** [review]
>> I never claimed Windows didn't support IPC I clearified Rust in my PR intro `Certain parts of Trin either won't compile on Windows or crash because rust doesn't support windows Unix IPC` --> `rust doesn't support windows Unix IPC`

I remember now the discussion about **Tokio** and windows uds. By claim, I was referring to the message in this panic: https://github.com/ethereum/trin/pull/923/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R34.

It was weird because it seems Rust [supports](https://crates.io/crates/uds_windows) windows Unix IPC, it is **Tokio** that has some issues with it.  So I guess fixing the PR description and the panic will make it less confusing?

**Comment by KolbyML** [comment]
That is the library we used to use 2 years ago.

**Comment by KolbyML** [comment]
Oh that panic error. Oh ok my bad I will fix that. I should really read panic errors I steal from my old PRs. Thank you for bringing it to light.

**Comment by KolbyML** [comment]
Ok I resolved the panic error issue

**Review by carver — APPROVED** [review]
LGTM 🚢 

👍🏻 Yeah, switching from build to check can go in another PR.
